agent/api/statechange_test.go

```diff
@@ -361,3 +362,57 @@ func TestNewManagedAgentChangeEvent(t *testing.T) {
 		})
 	}
 }
+
+// TestTaskStateChangeMetadataGetter validates the data stored in
+// TaskMetadataGetter to check its implementation. The Getter should
+// fetch this data from the task reference, not the state change itself.
+func TestTaskStateChangeMetadataGetter(t *testing.T) {
+	taskArn := "arn:123"
+	t1 := time.Now()
+	t2 := t1.Add(1)
+	t3 := t2.Add(1)
+	task := &apitask.Task{
+		Arn:                      taskArn,
+		SentStatusUnsafe:         apitaskstatus.TaskRunning,
+		PullStartedAtUnsafe:      t1,
+		PullStoppedAtUnsafe:      t2,
+		ExecutionStoppedAtUnsafe: t3,
+	}
+
+	metadataGetter := NewTaskMetadataGetter(task)
+
+	change := &ecs.TaskStateChange{
+		MetadataGetter: metadataGetter,
+	}
+
+	assert.NotNil(t, change.MetadataGetter)
+	assert.Equal(t, change.MetadataGetter.GetTaskIsNil(), false)
```

Comment:
The last 2 arguments should be swapped here, right? (i.e., `false` is the expected and `change.MetadataGetter.GetTaskIsNil()` is the actual)

Same comment applies for all other lines calling `assert.Equal`

Reply:
ack